src/components/implementation/no_interface/llbooter/llbooter.c

```diff
@@ -344,18 +429,64 @@ execute(void)
 	crt_compinit_execute(boot_comp_get);
 }
 
+void
+init_done_chkpt(struct crt_comp *c)
+{
+	struct crt_comp  *new_comp = boot_comp_get(crt_ncomp() + 1);
+	struct crt_chkpt *chkpt;
+	thdcap_t          thdcap;
+	int               ret;
+	char              name[32];
+
+	if (c->id == cos_compid()) {
+	 	/* don't allow chkpnts of the booter */
+	 	BUG();
+	}
+
+	memcpy(name, "chkpt_", sizeof("chkpt_"));
+	assert(strlen(name) < strlen("chkpt_") + strlen(c->name));
```

Comment:
I don't think that all of the logic around the strings makes sense here.

1. I'm not sure if line 446 is copying the `\0` at the end of the string.
2. this assertion is always true, right? strlen(name) is identical to strlen("chkpt_").
3. the strcat might as well do the right thing despite the assertion check above. If you remove the assertion then 448 can overrun `name`. Assertions are checks, and we shouldn't really rely on their correctness.

All of this can be replaced with

```c
char *prefix = "chkpt_";
int prefix_sz = strlen("chkpt_");
...
assert(INITARGS_MAX_PATHNAME > prefix_sz + strlen(c->name));
memcpy(name, prefix, prefix_sz + 1);  //check my logic, but I think we want to copy the \0
strncat(name, c->name, INITARGS_MAX_PATHNAME - prefix_sz - 1);
c->name[INITARGS_MAX_PATHNAME - 1] = '\0';
```

Reply:
Most important, just make sure that you test it, but you might want to write a small test on Linux. Not sure.